### PR TITLE
python3Packages.chainer: 6.5.0 -> 7.7.0

### PR DESCRIPTION
**Pull request by ishiy1993**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ishiy1993** [comment]
```
2 packages marked as broken and skipped:
python37Packages.optuna python38Packages.optuna

3 packages built:
python37Packages.chainer python38Packages.chainer python39Packages.chainer
```

File: pkgs/development/python-modules/chainer/default.nix [hunk 1]
```diff
@@ -35,7 +30,7 @@ buildPythonPackage rec {
 
   # avoid gpu tests
   checkPhase = ''
-    pytest tests/chainer_tests/utils_tests -k 'not gpu and not cupy'
+    pytest tests/chainer_tests/utils_tests -k 'not gpu and not cupy and not ideep'
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/chainer/default.nix:33]: Please use pytestCheckHook and disabledTests.

Comment (ishiy1993) [reply pkgs/development/python-modules/chainer/default.nix]: I fix it. https://github.com/NixOS/nixpkgs/pull/111021/commits/c355637b4f182379eb9ffa3d9c492d3fb0b2dd74

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111021` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.optuna</li>
<li>python38Packages.optuna</li>
</ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.chainer</li>
<li>python38Packages.chainer</li>
<li>python39Packages.chainer</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111021` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.optuna</li>
<li>python38Packages.optuna</li>
</ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.chainer</li>
<li>python38Packages.chainer</li>
<li>python39Packages.chainer</li>
</ul>
</details>

**Pull request merged**